File: solutions_5756407898963968_0/C++/jasbir/g1.cpp
#include<cstdio>
using namespace std;
int find_common(int a[][4],int b[][4],int c,int d)
{
	int ans=-1;
	int count=0;
	for(int i=0;i<4;i++)
	for(int j=0;j<4;j++)	
	if(a[c][i]==b[d][j])
	{count ++;ans=a[c][i];
	}
   
	if(count ==1) return ans;
	if(count ==0) return -1;
	if(count>1) return -2;
}
int main()
{
	int ntc;
	int a[4][4],b[4][4];
	int x,y;
	scanf("%d",&ntc);
	for(int tc=1;tc<=ntc;tc++)
	{
		printf("Case #%d: ",tc);
		scanf("%d",&x);
		for(int i=0;i<4;i++)
		for(int j=0;j<4;j++)
		scanf("%d",&a[i][j]);
		scanf("%d",&y);
		for(int i=0;i<4;i++)
		for(int j=0;j<4;j++)
		scanf("%d",&b[i][j]);
		int ans=find_common(a,b,x-1,y-1);
		if(ans>0) printf("%d\n",ans);
		if(ans==-1) printf("Volunteer cheated!\n");
		if(ans==-2) printf("Bad magician!\n");
	}
}

File: solutions_5756407898963968_0/Python/Sriuba/g1.py
# input
fi = open("python/input.txt", "r")
lines = fi.readlines()
fi.close()

# main part
fo = open("python/output.txt", "wb")
t = int(lines[0])
for j in range(0, t):
  first_choice = int(lines[1 + 10*j])
  first_line = [int(i) for i in lines[1 + first_choice + 10*j].split(' ')]
  second_choice = int(lines[6 + 10*j])
  second_line = [int(i) for i in lines[6 + second_choice + 10*j].split(' ')]
  in_common = list(set(first_line).intersection(set(second_line)))
  if len(in_common) == 1:
    fo.write(bytes('Case #' + str(j+1) + ': ' + str(in_common[0]) + '\n', 'UTF-8'))
  elif len(in_common) == 0:
    fo.write(bytes('Case #' + str(j+1) + ': ' + 'Volunteer cheated!'+ '\n', 'UTF-8'))
  else:
    fo.write(bytes('Case #' + str(j+1) + ': ' + 'Bad magician!' + '\n','UTF-8'))
fo.close()
